app/workers/reduce_worker.rb

```diff
@@ -8,7 +8,16 @@ class ReduceWorker
   end
 
   def self.unique_args(args)
-    [args[0], args[1], args[2]]
+    reducible_id, reducible_class, subject_id, user_id, extract_ids = args
+    reducible = reducible_class.constantize.find(reducible_id)
+    reducers = reducible.reducers
+
+    uniques = [reducible_id, reducible_class]
+    uniques.push subject_id if reducers.any? { |r| r.reduce_by_subject? }
```

Comment:
Could these tests be pushed into sql instead of enumerating all the reducers each time? Maybe 
`Reducer.where(reducible_id: reducible_id, reducible_type: reducible_class, topic: 0).exists?`

Reply:
Is this approach actually faster? I'm happy to make the change if it is, I just stuck with simplicity for readability's sake